servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/AbstractSslConfigBuilder.java

```diff
@@ -0,0 +1,317 @@
+     * @return {@code this}.
+     */
+    public T alpnProtocols(final List<String> protocols) {
+        if (protocols.isEmpty()) {
+            throw new IllegalArgumentException("protocols cannot be empty");
+        }
+        this.alpnProtocols = protocols;
+        return thisT();
+    }
+
+    /**
+     * Set the TLS <a href="https://tools.ietf.org/html/rfc7301">ALPN</a> protocols.
+     * <p>
+     * Note that each ALPN protocol typically requires corresponding configuration at the protocol layer and as a result
+     * maybe inferred and overridden by the protocol layer.
+     * @param protocols the TLS <a href="https://tools.ietf.org/html/rfc7301">ALPN</a> protocols.
+     * @return {@code this}.
+     */
+    public T alpnProtocols(final String... protocols) {
+        return alpnProtocols(asList(protocols));
+    }
+
+    @Nullable
+    final List<String> alpnProtocols() {
+        return alpnProtocols;
+    }
+
+    /**
+     * Set the cipher suites to enable, in the order of preference.
+     *
+     * @param ciphers the ciphers to use.
+     * @return {@code this}.
+     */
+    public T ciphers(final List<String> ciphers) {
+        if (ciphers.isEmpty()) {
+            throw new IllegalArgumentException("ciphers cannot be empty");
+        }
+        this.ciphers = ciphers;
+        return thisT();
+    }
+
+    /**
+     * Set the cipher suites to enable, in the order of preference.
+     *
+     * @param ciphers the ciphers to use.
+     * @return {@code this}.
+     */
+    public T ciphers(final String... ciphers) {
+        return ciphers(asList(ciphers));
+    }
+
+    @Nullable
+    final List<String> ciphers() {
+        return ciphers;
+    }
+
+    /**
+     * Get the size of the cache used for storing SSL session objects.
+     *
+     * @param sessionCacheSize the size of the cache used for storing SSL session objects.
+     * @return {@code this}.
+     * @see SSLSessionContext#setSessionCacheSize(int)
+     */
+    public T sessionCacheSize(long sessionCacheSize) {
+        this.sessionCacheSize = sessionCacheSize;
+        return thisT();
+    }
+
+    final long sessionCacheSize() {
+        return sessionCacheSize;
+    }
+
+    /**
+     * Get the timeout for the cached SSL session objects, in seconds.
+     *
+     * @param sessionTimeout the timeout for the cached SSL session objects, in seconds.
+     * @return {@code this}.
+     * @see SSLSessionContext#setSessionTimeout(int)
+     */
+    public T sessionTimeout(long sessionTimeout) {
+        this.sessionTimeout = sessionTimeout;
+        return thisT();
+    }
+
+    final long sessionTimeout() {
+        return sessionTimeout;
+    }
+
+    /**
+     * Get the {@link SslProvider} to use.
+     *
+     * @param provider the {@link SslProvider} to use.
+     * @return {@code this}.
+     */
+    public T provider(SslProvider provider) {
+        this.provider = provider;
```

Comment:
`requireNonNull`?

Reply:
sure I can add it here. `null` is the default state so folks could conceivably "set back to default" ... but we can allow this later if we need to.